File: DialedInCoffee/Domain/HistorySnapshot.swift
import Foundation

public enum AdjustmentOutcome: String, Codable, CaseIterable, Hashable, Sendable {
    case unknown
    case better
    case worse
    case good

    public var displayName: String {
        switch self {
        case .unknown: return "Not rated"
        case .better: return "Better"
        case .worse: return "Worse"
        case .good: return "Good (last good)"
        }
    }
}

public struct HistorySnapshot: Codable, Hashable, Sendable, Identifiable {
    public var id: UUID
    public var date: Date
    public var beforeSettings: MachineSettings
    public var afterSettings: MachineSettings
    public var symptoms: [Symptom]
    public var primaryAdjustmentParameter: AdjustmentParameter
    public var outcome: AdjustmentOutcome
    public var beanName: String?
    public var drink: DrinkType

    public init(
        id: UUID = UUID(),
        date: Date = Date(),
        beforeSettings: MachineSettings,
        afterSettings: MachineSettings,
        symptoms: [Symptom],
        primaryAdjustmentParameter: AdjustmentParameter,
        outcome: AdjustmentOutcome = .unknown,
        beanName: String? = nil,
        drink: DrinkType
    ) {
        self.id = id
        self.date = date
        self.beforeSettings = beforeSettings
        self.afterSettings = afterSettings
        self.symptoms = symptoms
        self.primaryAdjustmentParameter = primaryAdjustmentParameter
        self.outcome = outcome
        self.beanName = beanName
        self.drink = drink
    }
}

File: android/core/src/main/kotlin/com/cawfee/domain/model/HistorySnapshot.kt
package com.cawfee.domain.model

import java.util.UUID

/** Outcome of a prior adjustment. Ported from HistorySnapshot.swift. */
enum class AdjustmentOutcome(val displayName: String) {
    UNKNOWN("Not rated"),
    BETTER("Better"),
    WORSE("Worse"),
    GOOD("Good (last good)");
}

/** A historical adjustment record. Ported from HistorySnapshot.swift. */
data class HistorySnapshot(
    val id: UUID = UUID.randomUUID(),
    /** Epoch milliseconds (replaces Swift Date for platform independence). */
    val dateMillis: Long = System.currentTimeMillis(),
    val beforeSettings: MachineSettings,
    val afterSettings: MachineSettings,
    val symptoms: List<Symptom>,
    val primaryAdjustmentParameter: AdjustmentParameter,
    val outcome: AdjustmentOutcome = AdjustmentOutcome.UNKNOWN,
    val beanName: String? = null,
    val drink: DrinkType,
)
